### PR TITLE
Add person student and teacher classes for the school library project

### PR DESCRIPTION
**Pull request by lenkon**
#### The following implementations are made:
- Set up project
- Create Person class
- Create Student class
- Create Teacher class

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
```
Comment (julie-ify) [inline person.rb]: - [ ] Only the `of_age?` method should be private. Kindly move the second method to be public.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require_relative person
+
+class Student < Person
+  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission: true)
+    @classroom = classroom
+    @parent_permission = parent_permission
+  end
```
Comment (julie-ify) [inline student.rb]: - [ ] There is no use in stating `@parent_permission = parent_permission`. Let's avoid unnecessary assignments, kindly remove it.

- [ ] In the `super` method the `parent_permission` should take the supplied one.

  ```
  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @classroom = classroom
  end
  ```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+class Teacher < Person
+  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission: true)
+    @specialization = specialization
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    true
+  end
```
Comment (julie-ify) [inline teacher.rb]: - [ ] There is no use in stating `@parent_permission = parent_permission`. Let's avoid unnecessary assignments, kindly remove it.

- [ ] In the `super` method the `parent_permission` should take the supplied one.

  ```
  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @specialization = specialization
  end
  ```

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require_relative person
```
Comment (julie-ify) [inline teacher.rb]: - [ ] Kindly import the person class correctly.

  ```
  require_relative './person'
  ```

File: student.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+require_relative person
```
Comment (julie-ify) [inline student.rb]: - [ ] Kindly import the person class correctly.

  ```
  require_relative './person'
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @lenkon,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (lenkon) [reply person.rb]: Hi @julie-ify,

Thanks for the feedback. I've made the necessary changes.

Best Regards,
K. Inoulen Shanongba

File: student.rb (hunk 5 above)
Comment (lenkon) [reply student.rb]: Hi @julie-ify,

Thanks for the feedback. I have made the required changes. But, the path gets changed to 'person' eg: require_relative 'person' after running rubocop automatic correction.


Best Regards,
K. Inoulen Shanongba

File: student.rb (hunk 2 above)
Comment (lenkon) [reply student.rb]: Hi @julie-ify,

Thanks for your valuable feedback. I have updated the code accordingly.

Best Regards,
K. Inoulen Shanongba

File: teacher.rb (hunk 4 above)
Comment (lenkon) [reply teacher.rb]: Hi @julie-ify,

Thanks for the feedback. I have updated the code as suggested. But when running the Rubocop autocorrect the path changes to 'person'

Best Regards,
K. Inoulen Shanongba

File: teacher.rb (hunk 3 above)
Comment (lenkon) [reply teacher.rb]: Hi @julie-ify,

Thanks for the thorough feedback. I've made the necessary changes.

Best Regards,
K. Inoulen Shanongba

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️

**Hi @lenkon,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**